src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs

```diff
@@ -342,25 +342,21 @@ private void TasView_ColumnClick(object sender, InputRoll.ColumnClickEventArgs e
 				}
 				else if (columnName != CursorColumnName)
 				{
-					var frame = TasView.AnyRowsSelected ? TasView.FirstSelectedRowIndex : 0;
 					var buttonName = TasView.CurrentCell.Column!.Name;
 
 					if (ControllerType.BoolButtons.Contains(buttonName))
 					{
 						if (ModifierKeys != Keys.Alt)
 						{
 							// nifty taseditor logic
-							bool allPressed = true;
-							foreach (var index in TasView.SelectedRows)
+							// your TAS Editor logic failed us (it didn't account for non-contiguous `SelectedRows`) --yoshi
+							var selection = TasView.SelectedRows.ToArray(); // sorted asc, length >= 1
+							var allPressed = selection.All(index => CurrentTasMovie.BoolIsPressed(index, buttonName))

```

Comment:
This throws if the last row is selected. The bounds check that comes after the `&&` here should be first.